### PR TITLE
api-examples: add ruby list_nodes.rb

### PR DESCRIPTION
**Pull request by dougsland**
Show how to use BlueChi and ruby programming language.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

**Comment by dougsland** [comment]
```
# dnf install ruby -y
# gem install ruby-dbus

# ruby ./list_nodes.rb
BlueChi nodes:
================
Name: control
Path: online
Status: /org/eclipse/bluechi/node/control

Name: node1
Path: offline
Status: /org/eclipse/bluechi/node/node1
```

File: doc/api-examples/ruby/list_nodes.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+bc_object = bc_service["/org/eclipse/bluechi"]
+bc_interface = bc_object["org.eclipse.bluechi.Manager"]
+nodes = bc_interface.ListNodes
+
+puts "BlueChi nodes:"
+puts "================"
+nodes.each do |node|
+  print "Name: ", node[0], "\n"
+  print "Path: ", node[2], "\n"
+  print "Status: ", node[1], "\n\n"
```
Comment (engelmi) [inline doc/api-examples/ruby/list_nodes.rb]:
```suggestion
  print "Path: ", node[1], "\n"
  print "Status: ", node[2], "\n\n"
```
These two should be switched - based on the [API description](https://github.com/containers/bluechi/blob/main/data/org.eclipse.bluechi.Manager.xml#L9), the path (`o`) is at position 1.

File: doc/docs/api/examples.md [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
-These steps were based on CentOS Stream 9:
+### Installing ruby and dbus module
 
-### Installing golang
+``` bash
+dnf install ruby -y
+gem install ruby-dbus
+```
+
+### Build the source code
```
Comment (engelmi) [inline doc/docs/api/examples.md:15]: Isn't this section more like `Build and run the example code`?

File: doc/docs/api/examples.md [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
+```
+
+---
+
+## GoLang
+
+### Installing golang and dbus module
 
 ``` bash
+# dnf install go-toolset
```
Comment (engelmi) [inline doc/docs/api/examples.md:43]: What exactly is this package needed for?

**Review by engelmi — COMMENTED** [review]
Nice! Would be great to also have more examples for ruby, basically all we have already for python. Could you add those in a later PR?

File: doc/docs/api/examples.md (hunk 3 above)
Comment (dougsland) [reply doc/docs/api/examples.md:43]: looks like as I added some lines above golang session it changed in the markdown?

File: doc/docs/api/examples.md (hunk 2 above)
Comment (dougsland) [reply doc/docs/api/examples.md:15]: sure, works for me.

File: doc/api-examples/ruby/list_nodes.rb (hunk 1 above)
Comment (dougsland) [reply doc/api-examples/ruby/list_nodes.rb]: you meant, change lines? like line 13 will be status and 14 Path? or change the position in the array ? We are using a library in ruby, not sure if I can do that without changing the library?

Comment (engelmi) [reply doc/api-examples/ruby/list_nodes.rb]: Like it is in the diff: The node path should be at array position 1, the status at position 2.

**Comment by dougsland** [comment]
> Nice! Would be great to also have more examples for ruby, basically all we have already for python. Could you
> add those in a later PR?

Of course! Give me few days 🚀

**Comment by dougsland** [comment]
Closing  in favor of a fresh PR: https://github.com/containers/bluechi/pull/536

**Pull request closed without merging**